docs/vector/geojson-support.md

```diff
@@ -1,59 +1,89 @@
+
+This is all necessary underpinning, but note that it is generally desirable to (de)serialize collections of features.  The serialization can be achieved by calling `.toGeoJson` on a `Seq[Feature[G, T]]`.  The result is a Json representation of a FeatureCollection.
+
+## Deserializing from GeoJson ##
+
+The return trip from a string representation can be accomplished by another method extension provided for strings: `parseGeoJson[T]`.  The only requirement for using this method is that the type of `T` must match the contents of the Json string.  If the Json string represents some `Geometry` subclass (i.e., `Point`, `MultiPolygon`, etc), then that type should be supplied to `parseGeoJson`.  This will work to make the return trip from any of the Json strings produced above.
+
+Again, it is generally more interesting to consider Json strings that contain `FeatureCollection` structures.  These require more complex code.  Consider the following Json string:
 ```Scala
-val jsonFeatureMapping = jsonString.parseGeoJson[geotrellis.vector.io.json.JsonFeatureCollectionMap]
+val fc: String = """{
+                   |  "type": "FeatureCollection",
+                   |  "features": [
+                   |    {
+                   |      "type": "Feature",
+                   |      "geometry": { "type": "Point", "coordinates": [1.0, 2.0] },
+                   |      "properties": { "SomeProp": 14 },
+                   |      "id": "target_12a53e"
+                   |    }, {
+                   |      "type": "Feature",
+                   |      "geometry": { "type": "Point", "coordinates": [2.0, 7.0] },
+                   |      "properties": { "SomeProp": 5 },
+                   |      "id": "target_32a63e"
+                   |    }
+                   |  ]
+                   |}""".stripMargin
+```
+Decoding this structure will require the use of either `JsonFeatureCollection` or `JsonFeatureCollectionMap`; the former will return queries as a `Seq[Feature[G, T]]`, while the latter will return a `Map[String, Feature[G, T]]` where the key is the `id` field of each feature.  After calling
+```scala
+val collection = fc.parseGeoJson[JsonFeatureCollectionMap]
 ```
+it will be necessary to extract the desired features from `collection`.  In order to maintain type safety, these results are pulled using accessors such as `.getAllPoints`, `.getAllMultiLineFeatures`, and so on.  Each geometry and feature type requires the use of a different method call.
 
-2. Once the string (which must be valid geojson!) is parsed, it is
-time to query your JsonFeatureCollection for its points:
+As in the case of serialization, to extract the feature data from this example string, we must create a case class with an integer member named `SomeProp` and register it using `jsonFormat1`.
+```scala
+case class SomeProp(SomeProp: Int)
```

Comment:
It's non-idiomatic in JSON to have upper case property names. Can we change this here and in the JSON to `someProp`?